src/main/java/de/bonndan/nivio/IndexEvent.java

```diff
@@ -15,6 +16,7 @@
 
     private final String message;
     private final LandscapeDescription landscapeDescription;
+    private ProcessLog processLog;
```

Comment:
I don't fully understand how it should work. 

That the `ApiController` should return a JSON response containing the link to the log and to the landscape is fine. So this would be for `indexLandscape()` and `create()`, correct?

What needs to change because of the async stuff would be this method in `IndexEventListener`, right?
````
@Override
public void onApplicationEvent(IndexEvent event) {
    LandscapeDescription description = event.getLandscape();
    LOGGER.debug("Received index event for landscape {}", description.getIdentifier());
    event.setProcessLog(indexer.index(description)); 
}
````
How do we get a link to the `ProcessLog` and to the `Landscape` from there? And also before `indexer.index(description)` has been completed?

Regarding the inmutability: I had a look at that and did not really know how to implement an inmutable `IndexEvent`. The ProcessLog is only available after the processing has been completed, so we must set it through the setter. An option would be to not locate the ProcessLog to the event. But then, to what should it belong then?

Reply:
You got that right 👍 

The link to the process log is always available, e.g. https://nivio-demo.herokuapp.com/api/landscape/nivio:example/log and also already documented in hateoas (https://nivio-demo.herokuapp.com/api/nivio:example). So just sending back a hateoas response is consistent and easy to implement.

Hopefully the events are already immutable (implicitly) and we should keep them this way.